crates/uopool/src/utils.rs

```diff
@@ -146,6 +157,91 @@ pub mod tests {
         assert_eq!(gas_oh.calculate_pre_verification_gas(&uo), 45340.into());
     }
 
+    #[test]
+    fn pre_verification_gas_calculation_with_large_user_operation() {
+        let gas_oh = Overhead::default();
+        let uo = UserOperation {
+            sender: "0xAB7e2cbFcFb6A5F33A75aD745C3E5fB48d689B54"
+                .parse()
+                .unwrap(),
+            nonce: U256::max_value(),
+            init_code: Bytes::from(vec![255; 1024]), // Large init_code
+            call_data: Bytes::from(vec![255; 1024]), // Large call_data
+            call_gas_limit: U256::max_value(),
+            verification_gas_limit: U256::max_value(),
+            pre_verification_gas: U256::max_value(),
+            max_fee_per_gas: U256::max_value(),
+            max_priority_fee_per_gas: U256::max_value(),
+            paymaster_and_data: Bytes::from(vec![255; 1024]), // Large paymaster_and_data
+            signature: Bytes::from(vec![255; 1024]),          // Large signature
+        };
+
+        assert_eq!(gas_oh.calculate_pre_verification_gas(&uo), 110020.into());
+    }
+
+    #[test]
+    fn pre_verification_gas_calculation_with_large_per_user_op_word() {
+        let gas_oh = Overhead {
+            fixed: U256::from(21000),
+            per_user_op: U256::from(18300),
+            per_user_op_word: U256::from(10000),
+            zero_byte: U256::from(4),
+            non_zero_byte: U256::from(16),
+            bundle_size: U256::from(1),
+            sig_size: U256::from(65),
+        };
+        let uo = UserOperation {
+            sender: "0xAB7e2cbFcFb6A5F33A75aD745C3E5fB48d689B54"
+                .parse()
+                .unwrap(),
+            nonce: U256::max_value(),
+            init_code: Bytes::from(vec![255; 1024]), // Large init_code
+            call_data: Bytes::from(vec![255; 1024]), // Large call_data
+            call_gas_limit: U256::max_value(),
+            verification_gas_limit: U256::max_value(),
+            pre_verification_gas: U256::max_value(),
+            max_fee_per_gas: U256::max_value(),
```

Comment:
It could be divided by zero if `max_fee_per_gas` is zero based on #191 . Would you mind also fixing it?

Reply:
ok! i will fix it

Reply:
Is it correct like this?
[`6e3b6cd` (#214)](https://github.com/Vid201/silius/pull/214/commits/6e3b6cde063f5a72c3d3ed6cbbc51f74f35eb3f8)